src/components/text-editor/prosemirror-adapter/menu/menu-commands.ts

```diff
@@ -1,93 +1,115 @@
+): Command => {
+    const type: NodeType | undefined = schema.nodes[nodeType];
+    if (!type) {
+        throw new Error(`Node type "${nodeType}" not found in schema`);
     }
 
-    private createToggleMarkCommand(markName: string) {
-        const markType: MarkType | undefined = this.schema.marks[markName];
-        if (!markType) {
-            throw new Error(`Mark "${markName}" not found in schema`);
-        }
+    if (nodeType === 'heading' && level) {
+        return setBlockType(type, { level: level });
+    } else {
+        return setBlockType(type);
+    }
+};
 
-        return toggleMark(markType);
+const createWrapInCommand = (schema: Schema, nodeType: string): Command => {
+    const type: NodeType | undefined = schema.nodes[nodeType];
+    if (!type) {
+        throw new Error(`Node type "${nodeType}" not found in schema`);
     }
 
-    private createSetNodeTypeCommand(nodeType: string, level?: number) {
-        const type = this.schema.nodes[nodeType];
-        if (!type) {
-            throw new Error(`Node type "${nodeType}" not found in schema`);
-        }
+    return wrapIn(type);
+};
 
-        if (nodeType === 'heading' && level) {
-            return setBlockType(type, { level: level });
-        } else {
-            return setBlockType(type);
-        }
+const createListCommand = (schema: Schema, listType: string): Command => {
+    const type: NodeType | undefined = schema.nodes[listType];
+    if (!type) {
+        throw new Error(`List type "${listType}" not found in schema`);
     }
 
-    private createWrapInCommand(nodeType: string) {
-        const type = this.schema.nodes[nodeType];
-        if (!type) {
-            throw new Error(`Node type "${nodeType}" not found in schema`);
-        }
+    return wrapInList(type);
+};
+
+const commandMapping: CommandMapping = {
+    strong: createToggleMarkCommand,
+    em: createToggleMarkCommand,
+    underline: createToggleMarkCommand,
+    headerlevel1: (schema) =>
+        createSetNodeTypeCommand(schema, 'heading', levelMapping.one),
+    headerlevel2: (schema) =>
+        createSetNodeTypeCommand(schema, 'heading', levelMapping.two),
+    headerlevel3: (schema) =>
+        createSetNodeTypeCommand(schema, 'heading', levelMapping.three),
+    blockquote: createWrapInCommand,
+    ordered_list: createListCommand,
+    bullet_list: createListCommand,
+    link: createToggleMarkCommand,
+};
+
+export class MenuCommandFactory {
+    private schema: Schema;
 
-        return wrapIn(type);
+    constructor(schema: Schema) {
+        this.schema = schema;
     }
 
-    private createListCommand(listType: string) {
-        const type = this.schema.nodes[listType];
-        if (!type) {
-            throw new Error(`List type "${listType}" not found in schema`);
+    getCommand(mark: EditorMenuTypes, url?: string) {
+        if (!Object.values(EditorMenuTypes).includes(mark)) {
+            throw new Error(`The Menu Type "${mark}" is not supported`);
+        }
```

Comment:
Is this really necessary? Can't we trust the consumers of `getCommand` to do type checking?

Reply:
We could definitely do that.